x-pack/test/apm_api_integration/basic/tests/services/service_details.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../../../../common/ftr_provider_context';
+import archives from '../../../common/archives_metadata';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const esArchiver = getService('esArchiver');
+
+  const archiveName = 'apm_8.0.0';
+  const range = archives[archiveName];
+  const start = encodeURIComponent(range.start);
+  const end = encodeURIComponent(range.end);
+
+  describe('Service details', () => {
+    describe('when data is not loaded ', () => {
+      it('handles the empty state', async () => {
+        const response = await supertest.get(
+          `/api/apm/services/opbeans-java/metadata/details?start=${start}&end=${end}&uiFilters=%7B%7`
```

Comment:
If we use `url.format` on these (we're doing it in some other API tests) we don't need to URI encode anything and it's easier to read.